www/js/controller.js

```diff
@@ -865,7 +865,8 @@ define([
 				console.log("Parser error: ", err);
 				console.log(err.message);
 				console.log(err.Error);
-				this._model.authored.setAuthorStatus(this.currentID,'equation','incorrect');
+				if(this._mode === "AUTHOR")
+					this._model.active.setAuthorStatus(this.currentID,'equation','incorrect');
```

Comment:
this check is already done in model.js. model.active gets authored with the mode is author else it gets student. You dont need to do this explicitly.